🌱  Updated Go to v1.23 and related k8s.io packages

Ankit152 opened the pull request:
<!--
Thanks for creating a pull request!

If this is your first time, please make sure to review CONTRIBUTING.MD.

Please copy the appropriate `:text:` or icon to the beginning of your PR title:

:sparkles: ✨ feature
:bug: 🐛 bug fix
:book: 📖 docs
:memo: 📝 proposal
:warning: ⚠️ breaking change
:seedling: 🌱 other/misc
:question: ❓ requires manual review/categorization

-->
## Summary
Updated Golang to `v1.23` and related `k8s.io` packages.

## Related issue(s)

Fixes https://github.com/open-cluster-management-io/ocm/issues/868

Ankit152 committed:
feat: updated go to v1.23 and related k8s.io packages

Signed-off-by: Ankit152 <ankitkurmi152@gmail.com>

Ankit152 commented:
Hey @haoqing0110 @zhujian7 please take a look into the PR. Thanks! 🙂

zhujian7 commented on client/addon/clientset/versioned/clientset.go:
```diff
@@ -3,8 +3,8 @@
 import (
-	"fmt"
-	"net/http"
+	fmt "fmt"
```
Is this auto generated?  Is it a new change of golang 1.23?

Ankit152 replied:
Yes. It was auto generated. 🙂

zhujian7 commented:
/ok-to-test

haoqing0110 commented:
/hold until we release v0.16.0

qiujian16 commented:
you will need to take a look at this PR https://github.com/open-cluster-management-io/api/pull/348/files#diff-2f46d2e309467f55ec125b43cb3efd00dd286ba5af10dba025b99ac261fda167 for the upgrade

We have to remove crd v1beta1 in this PR for upgrade.

Ankit152 commented:
Hey @qiujian16 I have made the changes. Kindly review them and I think this will require removing `do-not-merge/hold` to re-trigger the tests.

qiujian16 commented:
I think you would also need to update the makefile and some scripts under hack to make sure "make verify" can pass.

qiujian16 commented:
it is close, could you run
`make update`
`make verify` and ensure `make verify` can pass?

Looks like some CRD/clients are not correctly updated.

Ankit152 commented:
@qiujian16 I did ran the `make update` command earlier. But seems like some CRDs were broken and the changes were not populated correctly.
I have done that again and locally things are passing.

qiujian16 commented:
Integration fails, @xuezhaojun would you take a look why this field is required?

xuezhaojun commented:
@qiujian16  The fix PR submitted: https://github.com/open-cluster-management-io/api/pull/364

The root cause is that controller-gen 0.15.2 behive different from 0.16.5; For controller-gen 0.16.5, it marks kubeconfigSecrets as required, but for 0.15.2, it does not.

@Ankit152 Please rebase this PR after the fix PR is merged,  and `make update` again, that will solve the issue.

haoqing0110 commented:
/unhold

qiujian16 commented:
you might need to upgrade golang cilint version

Ankit152 commented:
Yes @qiujian16, I have updated the `golang-cilint` version to `v1.64.4` and fixed other lint issues. The CI should be fine now.
I will also be updating the lint version to `v1.64.6` to all the other repos as well. 🙂

qiujian16 commented:
/approve
/lgtm

Thanks! Great work.
